addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/WelcomeWebHookServlet.java

```diff
@@ -38,29 +32,35 @@
  *
  * @author Gaël L'hopital - Initial contribution
  */
-@Component(service = HttpServlet.class, configurationPolicy = ConfigurationPolicy.OPTIONAL, immediate = true)
 public class WelcomeWebHookServlet extends HttpServlet {
     private static final long serialVersionUID = 1288539782077957954L;
-    public static final String PATH = "/netatmo/camera";
+    private static final String PATH = "/netatmo/%s/camera";
     private static final String APPLICATION_JSON = "application/json";
     private static final String CHARSET = "utf-8";
     private final Gson gson = new Gson();
 
     private final Logger logger = LoggerFactory.getLogger(WelcomeWebHookServlet.class);
 
     private HttpService httpService;
-    private NetatmoHandlerFactory netatmoHandlerFactory;
+    private NetatmoBridgeHandler bridgeHandler;
+    private String path;
+
+    public WelcomeWebHookServlet(@NonNull HttpService httpService, String id) {
```

Comment:
In general we want to avoid using @nonnull but use @nonnullbydefault on class level.

See also https://www.eclipse.org/smarthome/documentation/development/guidelines.html#a-code-style A.12

Reply:
I tried to move the class WelcomeWebHookServlet to @NonNullByDefault but in this case, the overriden method service shows illegal redefinition of parameters.
By the way the discovery service is not "@NonDefaultByDefault" too.
Maybe you could rework the full annotations of the binding in a different PR.
Sorry but I forgive.

Reply:
Good point. If the superclass has no annotations it won't work. But there is no need for a @NonNull because it is set as default. If the `httpService` could be `null` we should prepare a check inside the constructor and throw a `RuntimeException` (e.g. an `IllegalArgumentException`)

Reply:
Done